p2p/src/secret_connection.rs

```diff
@@ -431,11 +475,11 @@ where
                 data_copy = &[0_u8; 0];
             }
             let sealed_frame = &mut [0_u8; TAG_SIZE + TOTAL_FRAME_SIZE];
-            let res = self.encrypt(chunk, sealed_frame);
+            let res = self.encrypt(chunk, &send_state.send_nonce, sealed_frame);
```

Comment:
We can do

```rust
self.encrypt(chunk, &send_state.send_nonce, sealed_frame)
  .map_err(|err| io::Error::new(io::ErrorKind::Other, err.to_string()))?;
```

With the `?` short circuiting the loop and returns the error result.